TACC/Core-Portal
# Bug/FP-1009: Apply ReadMore UI pattern to long Shared Workspace descriptions

## Overview: ##
Adds ReadMore UI pattern to long Shared Workspace descriptions.

## Related Jira tickets: ##

* [FP-1009](https://jira.tacc.utexas.edu/browse/FP-1009)

## Summary of Changes: ##
- Changed "Read More" button to left align rather than inheriting parent value. (`ReadMore.js`)
- Wrapped Shared Workspace description with ReadMore style.

## Testing Steps: ##
1. Go to a shared workspace.
2. Edit the description to a very long line so that it takes up more than 4 lines on a half-sized window.
3. Ensure that you see the "Read More" button with the last line truncated with a "...".
4. Ensure that the expand behavior works.
5. Ensure that when maximizing the screen (i.e. when the description reduces to less than 4 lines) the "Read More" button goes away.


## UI Photos:
![Peek 2021-07-01 16-19](https://user-images.githubusercontent.com/9425579/124190660-4603e580-da88-11eb-89eb-f4578de1ccf8.gif)


## Notes: ##
There is a related follow-up ticket regarding adding a character limit to descriptions and titles [FP-1110](https://jira.tacc.utexas.edu/browse/FP-1110).

I was trying to add a test that would check whether or not "Read More" shows up by providing it a long description and manipulating resize behavior. But apparently [you need Enzyme to test the library](https://github.com/maslianok/react-resize-detector#testing-with-enzyme-and-jest) (react-resize-detector). Would there be another way to arbitrarily resize the test window in a test?
